tests/test_gpu_examples.py

```diff
@@ -1878,6 +1879,101 @@ def test_causal_lm_training_aqlm(self):
             assert trainer.state.log_history[-1]["train_loss"] is not None
 
 
+@require_torch_gpu
+@require_hqq
+@unittest.skipUnless(
+    version.parse(importlib.metadata.version("transformers")) >= version.parse("4.36.1"),
+    "test requires `transformers>=4.36.1`",
+)
+class PeftHQQGPUTests(unittest.TestCase):
+    r"""
+    HQQ + peft tests
+    """
+
+    def setUp(self):
+        self.causal_lm_model_id = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.causal_lm_model_id)
+
+    def tearDown(self):
+        r"""
+        Efficient mechanism to free GPU memory after each test. Based on
+        https://github.com/huggingface/transformers/issues/21094
+        """
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    def _check_inference_finite(self, model, batch):
+        # try inference without Trainer class
+        training = model.training
+        model.eval()
+        output = model(**batch.to(model.device))
+        assert torch.isfinite(output.logits).all()
+        model.train(training)
+
+    @pytest.mark.single_gpu_tests
+    def test_causal_lm_training_hqq(self):
```

Comment:
Can we expand this test to also check DoRA, since you already implemented it:

```python
    @parameterized.expand([False, True])
    def test_causal_lm_training_hqq(self, use_dora):
```

and below, add `use_dora=use_dora` to `LoraConfig`.